File: modules/common/src/main/java/com/sygic/modules/common/di/module/SdkInitializationManagerModule.java
package com.sygic.modules.common.di.module;

import com.sygic.modules.common.initialization.manager.SdkInitializationManager;
import com.sygic.modules.common.initialization.manager.SdkInitializationManagerImpl;
import dagger.Module;
import dagger.Provides;

import javax.inject.Singleton;

@Module
public class SdkInitializationManagerModule {

    @Singleton
    @Provides
    SdkInitializationManager provideSdkInitializationManager() {
        return new SdkInitializationManagerImpl();
    }
}

File: modules/common/src/main/java/com/sygic/modules/common/di/SdkInitializationManagerModule.kt
package com.sygic.modules.common.di

import com.sygic.modules.common.initialization.manager.SdkInitializationManager
import com.sygic.modules.common.initialization.manager.SdkInitializationManagerImpl
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class SdkInitializationManagerModule {

    @Singleton
    @Provides
    fun provideSdkInitializationManager(): SdkInitializationManager = SdkInitializationManagerImpl()
}